c_src/Makefile

```diff
@@ -110,8 +118,19 @@ build_erl_jq: $(OBJECTS)
 
 $(JQSRC):
 	git clone -b jq-1.6-emqx --single-branch $(JQURL) $(JQSRC_DIR)
-
-$(LIBJQ_NAME): $(JQSRC)
+	((cd $(JQSRC_DIR) && git checkout 51df83ccf4808867eee79a011c71a93a121be1b6) || \
+	     (echo "Failed to check out jq commit" && \
+	      rm -r $(JQSRC_DIR) && \
+	      false))
+
+$(RDSSRC):
+	git clone -b master --single-branch $(RDSURL) $(RDSSRC_DIR)
+	((cd $(RDSSRC_DIR) && git checkout 40cdd2544cfbd697ec9fa6d70fd971560cd9b4ff) || \
```

Comment:
I can try to fix that and let you know if I don't have permission to create forks under emqx. We could also discuss if its better to use a git submodule or include the source code directly in the repo (the same should be discussed for the jq library).

Reply:
> > A general question is I see jq_state struct has pointers in it, I guess you have checked that the data they point to are safe to get reused since the memcpy will copy the pointer as well.
> 
> As far as I have seen, the same jq_state struct can be reused to process multiple JSON objects. This is also tested in the concurrent_queries_test_ case (that reuses the same jq_state object many times due to the LRU cache). I don't think the jq_state struct is ever `memcpy'ed (if it was, it might be a problem). Please explain if I have misunderstood your concern.

After offline discussion, We have checked it is covered by unittest and memory sanitizer .

Reply:
Fixed in 687b57c below.